Prism/Prism/Classes/PrismURL.swift

```diff
@@ -0,0 +1,312 @@
+//
+//  PrismURL.swift
+//  prism-ios
+//
+//  Created by Göktuğ Berk Ulu on 21/09/2017.
+//  Copyright © 2017 Hipo. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public enum PrismOutputImageQuality {
+    case high
+    case normal
+    case low
+    case custom(quality: Int)
+    case none
+}
+
+public enum PrismOutputImageType {
+    case png
+    case jpg
+    case none
+}
+
+public enum PrismOutputImageResizeMode {
+    case resize
+    case fit
+    case crop
+    case none
+}
+
+public enum PrismOutputGravity {
+    case topLeft
+    case center
+    case none
+}
+
+public class PrismURL {
+    
+    var baseURL: URL? = nil
+    var quality: PrismOutputImageQuality = PrismOutputImageQuality.none
+    var expectedHeight: CGFloat = 0.0
+    var expectedWidth: CGFloat = 0.0
```

Comment:
The more appropriate solution may be to use CGSize instead of setting a width of height explicitly. If user wants to set one of them, then he/she should use something like 
       UIKIT_EXTERN const CGFloat UIViewNoIntrinsicMetric NS_AVAILABLE_IOS(6_0); // -1.
Also, I think 0 should be an invalid value. If the user sends it, we either return nil as prism URL or we should not add it as a query parameter(Unless prism API requires a value).

One more thing, if the user sets resize mode as none and a specific size, what should be the correct behaviour?

1- If resize mode has higher priority, we should not set size and return the original image.
2- If the size has higher priority, we should change none with a more appropriate word since none has no meaning for the concept.